File: frontend/src/api/account.ts
import request from './index'

// 注册账户
export const registerAccount = (data: {
  account_name: string
  email: string
  password: string
  company_name?: string
  contact_person?: string
  contact_phone?: string
}) => {
  return request.post('/account/register', data)
}

// 登录
export const login = (data: { email: string; password: string }) => {
  return request.post('/account/login', data)
}

// 查询余额
// 管理员模式下返回模拟数据，impersonate 模式调用真实 API
export const getBalance = () => {
  const isImpersonate = sessionStorage.getItem('impersonate_mode') === '1'
  if (isImpersonate) {
    return request.get('/account/balance')
  }
  const adminToken = localStorage.getItem('admin_token');
  if (adminToken) {
    return Promise.resolve({
      account_id: 'admin',
      balance: 0,
      currency: 'USD',
      low_balance_threshold: null
    });
  }
  return request.get('/account/balance')
}

/** 客户账户信息（/account/info） */
export interface AccountInfo {
  id: number
  account_name: string
  email?: string | null
  balance: number
  currency: string
  status: string
  services?: string
  company_name?: string | null
  contact_person?: string | null
  rate_limit?: number | null
  tg_id?: number | null
  tg_username?: string | null
  unit_price?: number | null
  created_at: string
  client_name?: string | null
  country_code?: string | null
  remaining_sms_estimate?: number | null
  sales_tg_username?: string | null
}

// 查询账户信息
// 管理员模式下返回模拟数据，impersonate 模式调用真实 API
export const getAccountInfo = (): Promise<AccountInfo> => {
  const isImpersonate = sessionStorage.getItem('impersonate_mode') === '1'
  if (isImpersonate) {
    return request.get('/account/info')
  }
  const adminToken = localStorage.getItem('admin_token');
  if (adminToken) {
    return Promise.resolve({
      id: 0,
      account_name: 'Admin',
      email: 'admin@system.local',
      balance: 0,
      currency: 'USD',
      status: 'active',
      company_name: 'System Administrator',
      contact_person: 'Admin',
      rate_limit: 0,
      created_at: new Date().toISOString()
    });
  }
  return request.get('/account/info')
}

/** 余额变动流水（仅资金事件：充值/退款/调整等，不含逐条 SMS 扣费） */
export interface BalanceTransaction {
  id: number
  type: string
  amount: number
  balance_after: number
  description: string
  created_at: string | null
}

// 近期交易（/account/transactions）
// 管理员非 impersonate 模式无真实账户上下文，返回空列表
export const getTransactions = (limit = 20): Promise<{ items: BalanceTransaction[] }> => {
  const isImpersonate = sessionStorage.getItem('impersonate_mode') === '1'
  if (!isImpersonate && localStorage.getItem('admin_token')) {
    return Promise.resolve({ items: [] })
  }
  return request.get('/account/transactions', { params: { limit } })
}

// 修改客户登录密码
export const changeAccountPassword = (data: { old_password: string; new_password: string }) => {
  return request.post('/account/change-password', data)
}

// 生成客户 TG 绑定码
export const generateAccountTgBindCode = () => {
  return request.post('/account/telegram-bind-code')
}

// 解绑客户 TG
export const unbindAccountTelegram = () => {
  return request.post('/account/telegram-unbind')
}

// 客户 TG 验证登录
export const sendAccountTelegramCode = (identifier: string) => {
  return request.post('/account/telegram-login/send-code', { identifier })
}

export const verifyAccountTelegramCode = (identifier: string, code: string) => {
  return request.post('/account/telegram-login/verify', { identifier, code })
}

File: backend/app/schemas/account.py
"""
账户相关的数据模式
"""
from typing import Optional
from pydantic import BaseModel, Field, EmailStr
from decimal import Decimal


class AccountBalanceResponse(BaseModel):
    """账户余额响应"""
    account_id: int
    balance: float
    currency: str
    low_balance_threshold: Optional[float] = None
    
    class Config:
        json_schema_extra = {
            "example": {
                "account_id": 1,
                "balance": 1250.50,
                "currency": "USD",
                "low_balance_threshold": 100.0
            }
        }


class AccountInfoResponse(BaseModel):
    """账户信息响应"""
    id: int
    account_name: str
    email: Optional[str] = None
    balance: float
    currency: str
    status: str
    services: str = "sms"
    company_name: Optional[str] = None
    contact_person: Optional[str] = None
    rate_limit: Optional[int] = None
    tg_id: Optional[int] = None
    tg_username: Optional[str] = None
    unit_price: Optional[float] = None  # 短信单价(USD/条)，用于前端预估费用
    created_at: str
    # 仪表盘等展示用
    client_name: Optional[str] = None  # 客户名称：优先公司名称，否则账户名
    country_code: Optional[str] = None  # 国家/地区代码
    remaining_sms_estimate: Optional[int] = None  # 按单价估算剩余可发条数（余额/单价向下取整）
    sales_tg_username: Optional[str] = None  # 归属商务 Telegram，便于客户联系


class AccountCreateRequest(BaseModel):
    """创建账户请求"""
    account_name: str = Field(..., min_length=2, max_length=100)
    email: EmailStr
    password: str = Field(..., min_length=8)
    company_name: Optional[str] = None
    contact_person: Optional[str] = None
    contact_phone: Optional[str] = None
    
    class Config:
        json_schema_extra = {
            "example": {
                "account_name": "测试公司",
                "email": "test@example.com",
                "password": "password123",
                "company_name": "Test Company Ltd.",
                "contact_person": "张三"
            }
        }


class AccountLoginRequest(BaseModel):
    """账户登录请求"""
    email: str  # 支持邮箱或用户名
    password: str


class AccountLoginResponse(BaseModel):
    """账户登录响应"""
    success: bool
    token: Optional[str] = None
    account_id: Optional[int] = None
    error: Optional[str] = None

